File: array_and_hashing/swift/isSubsequence.swift
func isSubsequence(_ s: String, _ t: String) -> Bool {
    if(s.count > t.count){
        return false
    }
    var sIdx = 0, tIdx = 0
    var s = Array(s), t = Array(t)
    while(sIdx < s.count && tIdx < t.count){
        if(s[sIdx] == t[tIdx]){
            sIdx+=1
        }
        tIdx+=1
    }
    return sIdx == s.count
}

File: array_and_hashing/cpp/isSubsequence.cpp
#include <string>

bool isSubsequence(std::string s, std::string t) {
    if (s.size() > t.size()){
        return false;
    }
    int s_i = 0, t_i = 0;

    while(s_i < s.size() && t_i < t.size()) {
        if(s[s_i] == t[t_i]){
            s_i++;
        }
        t_i++;
    }
    return s_i == s.size();
}